.github/scripts/hermetic_library_generation.sh

```diff
@@ -89,6 +89,10 @@ pushd "${api_def_dir}"
 git checkout "${googleapis_commitish}"
 popd
 
+# we also setup showcase
+source java-showcase/scripts/showcase_utilities.sh
+append_showcase_to_api_defs "${api_def_dir}"
```

Comment:
Can this function be called from `generate_showcase.sh` instead of here?

Reply:
It's called both from the automatic generation script (this one) and the manual update + golden test scripts (`generate_showcase.sh`). Since both scripts need to append the showcase protos to googleapis, I extracted this function into a utility file.

Reply:
My concern is that this `hermetic_library_generation.sh` is reused by google-cloud-java and all handwritten libraries, they don't need to copy showcase protos to googleapis. So can we move this step to a sdk-platform-java specific place? For example, maybe in the [workflow](https://github.com/googleapis/sdk-platform-java/blob/main/.github/workflows/hermetic_library_generation.yaml)?

Reply:
Ok, since the workflow also ends up using the common-usage `action.yaml`, I modified the scripts to accept a new flag `showcase_mode` that defaults to `false`. When `true`, it will download the api definitions. The download operation is still done in `hermetic_library_generation.sh`, but we prevent it from happening in repos that don't need these definitions.

Reply:
I think only [action.yaml](https://github.com/googleapis/sdk-platform-java/blob/main/.github/scripts/action.yaml) is reused. The `hermetic_library_generation.yaml` though is different in each repo. This is the one in [sdk-platform-java](https://github.com/googleapis/sdk-platform-java/blob/main/.github/workflows/hermetic_library_generation.yaml) and this is the one in [google-cloud-java](https://github.com/googleapis/google-cloud-java/blob/main/.github/workflows/hermetic_library_generation.yaml). So we could add the logic that appends showcase protos to the `hermetic_library_generation.yaml` in sdk-platform-java only, and it would not affect other repos.

Reply:
Done!